cpp/src/DataStorm/SessionI.cpp

```diff
@@ -743,8 +755,14 @@ SessionI::retry(NodePrx node, exception_ptr exception)
             return false;
         }
 
-        _retryTask = make_shared<IceInternal::InlineTimerTask>([node = std::move(node), self = shared_from_this()]
-                                                               { self->reconnect(node); });
+        _retryTask = make_shared<IceInternal::InlineTimerTask>(
+            [node = std::move(node), self = shared_from_this()]
+            {
+                self->_nextRetryAttemptTask =
+                    make_shared<IceInternal::InlineTimerTask>([self, node] { self->retry(node, nullptr); });
+                self->_instance->scheduleTimerTask(self->_nextRetryAttemptTask, 10s);
```

Comment:
I think we should add a separate property for this timeout. Not clear we will want exponential backoff strategy like we have for retries.